File: CCC-school-work/comporg/pnc2/mchon/js/primeregbs.js
#!/usr/bin/nodejs

function main()
{
	/////////////////////////////////////////////////////////////////////////////////
	//
	// Error checking
	//
	if (process.argv[2] == undefined || process.argv[2] == null)
	{
		console.error("Insufficient argument!");
		process.exit();
	}
	/////////////////////////////////////////////////////////////////////////////////
	//
	// Timing start
	//
	var start = new Date();
	/////////////////////////////////////////////////////////////////////////////////
	//
	// Variable declaration
	//
	var max = process.argv[2];
	var number = 2;
	var flag = 0;
	var qty = 0;
	
	/////////////////////////////////////////////////////////////////////////////////
	//
	// Main algorithm
	//
	while (qty < max)
	{
		for (var factor = 2; factor <= Math.sqrt(number); factor++)
		{
			if (number%factor == 0)
			{
				flag = 1;
				break;
			}
		}
		if (flag == 0)
		{
			process.stderr.write(number + " " );
			qty++;
		}
		flag = 0;
		number++;
	}
	console.error();
	/////////////////////////////////////////////////////////////////////////////////
	//
	// Timing end
	//
	var end = new Date();
	/////////////////////////////////////////////////////////////////////////////////
	//
	// Print time
	//
	var time = end.getTime() - start.getTime();
	console.log(time/1000);
}
main();

File: CCC-school-work/comporg/pnc2/mchon/python/primeregbs.py
#!/usr/bin/python

import sys
import math
import time

########################################################
##
## Error checking
##
if (len(sys.argv) < 2):
	print "Insufficient arguments!"
	sys.exit()

start = float(time.time())
########################################################
##
## Variable declaration
##
factor = 2
flag = 0
number = 2
maxqty = int(sys.argv[1])
qty = 0

########################################################
##
## Main algorithm
##
while (qty < maxqty):

	while (factor <= math.sqrt(number)):

		if (number%factor == 0):
			flag = 1
			break
		factor = factor + 1

	if (flag == 0):
		sys.stderr.write("%d "%number)
		qty = qty+1

	number = number + 1
	flag = 0
	factor = 2
print ""
end = float(time.time())
sys.stdout.write("%8.6f\n"%((end-start)))


